utils/validate_utils.py

```diff
@@ -2,82 +2,167 @@
+    cell_line : str
+        cell line metadata to add to correlations
 
     Returns
     -------
     pd.DataFrame
-        classification profiles dataframe
+        wide format dataframe with correlation info
     """
 
-    cell_line_classification_profiles = []
-
-    for cell_line in cell_line_plates:
-        # create one large dataframe for all 3 plates in the particular cell line
-        cell_line_plate_names = cell_line_plates[cell_line]
-        cell_line_plate_classifications = []
-        for cell_line_plate_name in cell_line_plate_names:
-            plate_classifications_link = f"{plate_classifications_dir_link}/{cell_line_plate_name}_cell_classifications.csv.gz"
-            plate_classifications = pd.read_csv(
-                plate_classifications_link, compression="gzip", index_col=0
-            )
-            cell_line_plate_classifications.append(plate_classifications)
-        cell_line_plate_classifications = pd.concat(
-            cell_line_plate_classifications, axis=0
-        ).reset_index(drop=True)
-
-        # create dataframe with cell classifications averaged across perturbation, include cell line metadata
-        # add cell line metadata
-        cell_line_plate_classifications["Metadata_cell_line"] = cell_line
-        # rename perturbation column to match the format of cell health label profiles, in this case "perturbation" corresponds to "reagent" because DeepProfiler (used much earlier in pipeline) makes no distinction
-        cell_line_plate_classifications = cell_line_plate_classifications.rename(
-            columns={"Metadata_Reagent": "Metadata_pert_name"}
+    # remove metadata columns
+    cleaned_final_profile_dataframe = final_profile_dataframe.drop(
+        columns=["Metadata_profile_id", "Metadata_pert_name", "Metadata_cell_line"]
+    )
+
+    # get correlations depending on type
+    if corr_type == "pearson":
+        corrs = cleaned_final_profile_dataframe.corr(method="pearson")
+    if corr_type == "ccc":
+        corrs = ccc(cleaned_final_profile_dataframe)
+        corrs = squareform(corrs)
+        np.fill_diagonal(corrs, 1.0)
+        corrs = pd.DataFrame(
+            corrs,
+            index=cleaned_final_profile_dataframe.columns.tolist(),
+            columns=cleaned_final_profile_dataframe.columns.tolist(),
         )
 
-        # get rid of extra metadata columns
-        phenotypic_classes = [
-            col
-            for col in cell_line_plate_classifications.columns.tolist()
-            if ("Metadata" not in col)
-            or (col not in ["Location_Center_X", "Location_Center_Y"])
-        ]
-        columns_to_keep = [
-            "Metadata_pert_name",
-            "Metadata_cell_line",
-        ] + phenotypic_classes
-        cell_line_plate_classifications = cell_line_plate_classifications[
-            columns_to_keep
+    # refactor correlations to easily readable format (from 70 cell health indicators)
+    corrs = corrs.iloc[70:, :70]
```

Comment:
Hmmmm you are right that this is fragile. The correlation dataframes are output as mirrored values (same correlations for `cc_all_high_h2ax` and `apoptosis` vs `apoptosis` and `cc_all_high_h2ax` for example). This line fomats the dataframe as we need it to create the clustered heatmap. It transforms the mirrored output from the `.corr()` method to a dataframe with phenotypic classes as the indices and cell health indicator as columns with the correlation values as values.

In this case we use `iloc[70:, :70]` because for each correlation dataframe there are 70 cell health indicators. I think that the only way to derive this with code is to look for columns/indices that begin with `cc_` or `vb_` but im not sure that would make the code much more flexible. But, it could make the code more readable/understandable.

What do you think?

Reply:
After more thought, I think hardcoding with `70` here is reasonable given that the number of cell health indicators should not be changing for this dataset in the future.